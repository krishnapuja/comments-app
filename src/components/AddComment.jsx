import React from "react";

const initialFormState = {name: '', message: ''};


const AddComment = ({handleModalClose, addCommentModalDisplay}) => {
    const [formState, setFormState] = React.useState(initialFormState);
    
    // handles onChange for all form fields
    const handleChange = (e) => {
        setFormState({...formState, [e.target.name]: e.target.value});
    };

    // creates a comment on click of add comment button
    const handleClick = async (e) => {
        e.preventDefault();
        await fetch('/createComment', {
            method: 'POST',
            // api headers to send json data and accept same
            headers: {
                'Accept': 'application/json, text/plain, */*',
                'Content-Type': 'application/json'
              },
            body: JSON.stringify(formState), 
          }).then(res => res.json())
        .then(res => {
            console.log(res);
        })
        .catch(e=>{
            console.log(e);
        })
    };

    // clears the modal window and clears form data on closing Modal
    const handleModalCloseClick = () => {
        setFormState(initialFormState);
        handleModalClose();
    }

    return (
        <main className="main-section">
            <div className="modal" style={{display: addCommentModalDisplay}}>
                <div className="modal-content">
                    <span className="close" onClick={handleModalCloseClick}>&times;</span>
                    <h3 className="add-comment-heading ">Add your Comment</h3>
                    <form className="margin-30">
                        <label htmlFor="name">Name</label><br />
                        <input type="text" id="name" name="name" 
                        className="input-field" 
                        value={formState.name} 
                        onChange={handleChange}/><br />
                        <label htmlFor="message">Message</label><br />
                        <textarea id="message" name="message"
                         rows="4" cols="50" 
                         className="input-field" 
                         value={formState.message} 
                         onChange={handleChange}></textarea> <br />
                         <div className="button-container">
                        <button onClick={handleClick} className="add-comment-button pad-10">Add Comment</button>
                        </div>
                    </form>
                </div>
            </div>
        </main>
    )
};

export default AddComment;