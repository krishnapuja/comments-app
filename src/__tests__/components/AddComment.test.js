import { render } from '@testing-library/react';
import AddComment from '../../components/AddComment';

describe('render AddComment correctly', () => {
  const mockProps = {
    handleModalClose: jest.fn(),
    addCommentModalDisplay: 'none',
  };

  test('renders the AddComment snapshot correctly', () => {
    const Apprenderer = render(<AddComment {...mockProps} />);
    expect(Apprenderer).toMatchSnapshot();
  });

  test('renders the AddComment snapshot correctly when modal is displayed', () => {
    mockProps.addCommentModalDisplay = 'block';
    const Apprenderer = render(<AddComment {...mockProps} />);
    expect(Apprenderer).toMatchSnapshot();
  });
});
