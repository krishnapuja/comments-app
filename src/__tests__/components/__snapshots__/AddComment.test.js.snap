// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AddComment correctly renders the AddComment snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main-section"
      >
        <div
          class="modal"
          style="display: none;"
        >
          <div
            class="modal-content"
          >
            <span
              class="close"
            >
              ×
            </span>
            <h3
              class="add-comment-heading "
            >
              Add your Comment
            </h3>
            <form
              class="margin-30"
            >
              <label
                for="name"
              >
                Name
              </label>
              <br />
              <input
                class="input-field"
                id="name"
                name="name"
                type="text"
                value=""
              />
              <br />
              <label
                for="message"
              >
                Message
              </label>
              <br />
              <textarea
                class="input-field"
                cols="50"
                id="message"
                name="message"
                rows="4"
              />
               
              <br />
              <div
                class="button-container"
              >
                <button
                  class="add-comment-button pad-10"
                >
                  Add Comment
                </button>
              </div>
            </form>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main-section"
    >
      <div
        class="modal"
        style="display: none;"
      >
        <div
          class="modal-content"
        >
          <span
            class="close"
          >
            ×
          </span>
          <h3
            class="add-comment-heading "
          >
            Add your Comment
          </h3>
          <form
            class="margin-30"
          >
            <label
              for="name"
            >
              Name
            </label>
            <br />
            <input
              class="input-field"
              id="name"
              name="name"
              type="text"
              value=""
            />
            <br />
            <label
              for="message"
            >
              Message
            </label>
            <br />
            <textarea
              class="input-field"
              cols="50"
              id="message"
              name="message"
              rows="4"
            />
             
            <br />
            <div
              class="button-container"
            >
              <button
                class="add-comment-button pad-10"
              >
                Add Comment
              </button>
            </div>
          </form>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`render AddComment correctly renders the AddComment snapshot correctly when modal is displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main-section"
      >
        <div
          class="modal"
          style="display: block;"
        >
          <div
            class="modal-content"
          >
            <span
              class="close"
            >
              ×
            </span>
            <h3
              class="add-comment-heading "
            >
              Add your Comment
            </h3>
            <form
              class="margin-30"
            >
              <label
                for="name"
              >
                Name
              </label>
              <br />
              <input
                class="input-field"
                id="name"
                name="name"
                type="text"
                value=""
              />
              <br />
              <label
                for="message"
              >
                Message
              </label>
              <br />
              <textarea
                class="input-field"
                cols="50"
                id="message"
                name="message"
                rows="4"
              />
               
              <br />
              <div
                class="button-container"
              >
                <button
                  class="add-comment-button pad-10"
                >
                  Add Comment
                </button>
              </div>
            </form>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main-section"
    >
      <div
        class="modal"
        style="display: block;"
      >
        <div
          class="modal-content"
        >
          <span
            class="close"
          >
            ×
          </span>
          <h3
            class="add-comment-heading "
          >
            Add your Comment
          </h3>
          <form
            class="margin-30"
          >
            <label
              for="name"
            >
              Name
            </label>
            <br />
            <input
              class="input-field"
              id="name"
              name="name"
              type="text"
              value=""
            />
            <br />
            <label
              for="message"
            >
              Message
            </label>
            <br />
            <textarea
              class="input-field"
              cols="50"
              id="message"
              name="message"
              rows="4"
            />
             
            <br />
            <div
              class="button-container"
            >
              <button
                class="add-comment-button pad-10"
              >
                Add Comment
              </button>
            </div>
          </form>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
